src/game.cpp

```diff
@@ -5419,6 +5419,7 @@ bool game::npc_menu( npc &who )
             u.mod_moves( -300 );
         }
     } else if( choice == sort_armor ) {
+        who.reset_encumbrance();
```

Comment:
You've added `reset_encumbrance` in three places.  Are these independent bugfixes or consequences of your change?  If the encumbrance is wrong we probably shouldn't be fixing it in these places in response to a UI action, we need to figure out what game mechanic altered the encumbrance and invalidate the encumbrance cache there.

Reply:
These are a consequence of picking up or dropping non-worn items not updating the encumbrance cache.

Reply:
Actually may have just been picking up, I need to do further testing.

Reply:
I'm not sure where I could put them in the drop/pickup code, as it seems to use different implementations in various places, not calling one standard thing.